matchms/typing.py

```diff
@@ -1,16 +1,13 @@
 from typing import List
-from typing import Optional
+from typing import NewType
 from typing import Tuple
 from typing import Union
 import numpy
-from .Spectrum import Spectrum
 
 
-SpectrumType = Optional[Spectrum]
+SpectrumType = NewType("Spectum", object)
```

Comment:
This was to avoid circular imports. I ran into issues because the Spectrum class was imported at places like this one before the actual Spectrum class definition got imported.

Reply:
Do you remember in which actual places? Circular imports are something that usually indicate that the interaction entities are somehow tightly couple and it could be beneficial to create another class which actually sits in between and contains the closely coupled code.

Reply:
I guess one of those places is seen here: https://github.com/matchms/matchms/runs/4659699326?check_suite_focus=true

```
_______________ ERROR collecting tests/test_add_compound_name.py _______________
ImportError while importing test module '/home/runner/work/matchms/matchms/tests/test_add_compound_name.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_add_compound_name.py:3: in <module>
    from matchms import Spectrum
matchms/__init__.py:1: in <module>
    from . import exporting
matchms/exporting/__init__.py:8: in <module>
    from .save_as_json import save_as_json
matchms/exporting/save_as_json.py:4: in <module>
    from ..Spectrum import Spectrum
matchms/Spectrum.py:7: in <module>
    from .filtering.add_precursor_mz import _add_precursor_mz_metadata
matchms/filtering/__init__.py:39: in <module>
    from .add_compound_name import add_compound_name
matchms/filtering/add_compound_name.py:2: in <module>
    from ..typing import SpectrumType
matchms/typing.py:6: in <module>
    from .Spectrum import Spectrum
E   ImportError: cannot import name 'Spectrum' from partially initialized module 'matchms.Spectrum' (most likely due to a circular import) (/home/runner/work/matchms/matchms/matchms/Spectrum.py)
```